lib/c-api/wasmer_deprecated.h

```diff
@@ -0,0 +1,1480 @@
+// The Wasmer C/C++ header file.
+
+#if !defined(WASMER_H_MACROS)
+
+// Include the WASP API for using the wasm_valkind_enum type.
+#include "wasm.h"
+
+#define WASMER_H_MACROS
+
+
+// Define the `ARCH_X86_X64` constant.
+#if defined(MSVC) && defined(_M_AMD64)
+#  define ARCH_X86_64
+#elif (defined(GCC) || defined(__GNUC__) || defined(__clang__)) && defined(__x86_64__)
+#  define ARCH_X86_64
+#endif
+
+// Compatibility with non-Clang compilers.
+#if !defined(__has_attribute)
+#  define __has_attribute(x) 0
+#endif
+
+// Compatibility with non-Clang compilers.
+#if !defined(__has_declspec_attribute)
+#  define __has_declspec_attribute(x) 0
+#endif
+
+// Define the `DEPRECATED` macro.
+#if defined(GCC) || defined(__GNUC__) || __has_attribute(deprecated)
+#  define DEPRECATED(message) __attribute__((deprecated(message)))
+#elif defined(MSVC) || __has_declspec_attribute(deprecated)
+#  define DEPRECATED(message) __declspec(deprecated(message))
+#endif
```

Comment:
Yes, perfect! It seems that we should be marking the entire deprecated API with `DEPRECATED("use X instead")` now?

Reply:
It'd be nice to have this macro available for our custom extensions to the Wasmer C API, we have 2 functions related to initializing WASI which are now deprecated and not marked like this, only marked by a comment